### DISCUSSION TITLE
build: error: C compiler cannot create executables (sandbox build)

### DISCUSSION THREAD

**Discussion by osevan** [netblue30/firejail#6417] [Q&A] [answered]
I'm trying to compile my apps inside a container.

I have success when my path is /root/compile.

But, when I compile under /usr/local/src/.... inside the container the C Compiler spits an error.

But config.log creates with success inside container.

```console
$ firejail --noprofile \
  --read-write=/usr/local/src/php$PHPVERSIONSHORT-build/php-$PHPVERSION \
  ./configure options...

firejail version 0.9.73

Parent pid 203116, child pid 203117
The new log directory is /proc/203117/root/var/log
Base filesystem installed in 1.17 ms
Child process initialized in 4.46 ms
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for cc... cc
checking whether the C compiler works... no
configure: error: in `/usr/local/src/php8.3-build/php-8.3.9':
configure: error: C compiler cannot create executables
See `config.log' for more details

Parent is shutting down, bye...
```

**Comment by rusty-snake** [comment]
Try with `--profile=noprofile.profile` and add basic debugging informations.

**Reply by osevan** [reply]
I will try and report back

**Comment by kmk3** [comment]
What are the names and versions of the distribution, compiler, libc and the
program that you're trying to build?

> I'm trying to compile my apps inside a container.

Why?

By "container" do you mean just inside of firejail or inside of an actual
container (like a docker container)?

> ```
> checking for cc... cc
> checking whether the C compiler works... no
> configure: error: in `/usr/local/src/php8.3-build/php-8.3.9':
> configure: error: C compiler cannot create executables
> See `config.log' for more details
> ```

What is the content of config.log?

**Reply by osevan** [reply]
I will show you tomorrow.

But compile and configure goes well - even inside fj container -,  when i try to compile inside /root/compile/.


I dont know why

**Reply by osevan** [reply]
> What are the names and versions of the distribution, compiler, libc and the program that you're trying to build?
> 
> > I'm trying to compile my apps inside a container.
> 
> Why?
> 
> By "container" do you mean just inside of firejail or inside of an actual container (like a docker container)?
> 
> > ```
> > checking for cc... cc
> > checking whether the C compiler works... no
> > configure: error: in `/usr/local/src/php8.3-build/php-8.3.9':
> > configure: error: C compiler cannot create executables
> > See `config.log' for more details
> > ```
> 
> What is the content of config.log?

Debian sid
Gcc 14...

With container i mean ofc fj container. 

Im not fan of podman/Docker snap or flatpaks

**Comment by osevan** [comment, marked as answer]
I resolved with fixing wrong variable inside cflags like `` removing on variable
